test/onnx/onnx_test_common.py

```diff
@@ -436,16 +436,18 @@ def _compare_pytorch_onnx_with_ort(
         ref_input_args = input_args
         ref_input_kwargs = input_kwargs
 
-    ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(
-        ref_model(*ref_input_args, **ref_input_kwargs)
-    )
-
+    # ONNXProgram holds a reference (not copy) to the original ref_model, including its state_dict.
+    # Thus, ONNXProgram() must run before ref_model() to prevent ref_model.forward() from changing the state_dict.
+    # Otherwise, the ref_model can change buffers on state_dict which would be used by ONNXProgram.__call__()
```

Comment:
I recall there was a PR that will make ONNXProgram.__call__() update original pytorch model lol. This problem will arise again if that's added.

Btw there is a flag `has_mutation` up there that triggers model cloning. Might be helpful.

Reply:
that pr was merged with this one.